components/script/dom/webglprogram.rs

```diff
@@ -144,6 +145,9 @@ impl WebGLProgram {
         }
 
         shader_slot.set(None);
+        // TODO: This is not complete because shaders can be attached to multple programs
+        // Was not sure were we should check if the shader is attached to other programs.
+        shader.set_is_attached(false);
```

Comment:
As the comment says this part is not complete.  Because we are using just a bool for the is_attached field and shaders can be attached to multiple programs I am not sure how you want to handle this.